api/types/role.go

```diff
@@ -472,18 +470,6 @@ func (r *RoleV3) CheckAndSetDefaults() error {
 		return trace.BadParameter("found invalid option in session_recording: %v", opt)
 	}
 
-	// if we find {{ or }} but the syntax is invalid, the role is invalid
-	for _, condition := range []RoleConditionType{Allow, Deny} {
-		for _, login := range r.GetLogins(condition) {
-			if strings.Contains(login, "{{") || strings.Contains(login, "}}") {
-				_, err := parse.NewExpression(login)
-				if err != nil {
-					return trace.BadParameter("invalid login found: %v", login)
-				}
-			}
-		}
-	}
-
```

Comment:
we should validate the role condition syntax on creation time.
will it still happen then, or will errors only show up when the role conditions are evaluated on user access?

Reply:
Errors will show up when attempting to upsert a role. Sounds reasonable?